### ISSUE TITLE
Cannot read properties of undefined (reading 'decorators')

### ISSUE THREAD

**Issue by khusamov** [typescript-rtti/typescript-rtti#56]
Rollup compile time error:

RTTI: During outboard metadata collection for class StopCommand: 
Cannot read properties of undefined (reading 'decorators')

RTTI: Failed to build source file khusamov-command-system/src/command/RepeatableCommand/StopCommand.ts: 
Cannot read properties of undefined (reading 'decorators') [please report]

```
TypeError: Cannot read properties of undefined (reading 'decorators')
    at MetadataEncoder.params (/Users/khusamov/Documents/repo/github.com/khusamov/solid-futurio-ru-game/node_modules/typescript-rtti/src/transformer/metadata-encoder.ts:299:39)
    at MetadataEncoder.class (/Users/khusamov/Documents/repo/github.com/khusamov/solid-futurio-ru-game/node_modules/typescript-rtti/src/transformer/metadata-encoder.ts:93:35)
    at /Users/khusamov/Documents/repo/github.com/khusamov/solid-futurio-ru-game/node_modules/typescript-rtti/src/transformer/metadata-emitter.ts:89:83
    at MetadataEmitter.collectMetadata (/Users/khusamov/Documents/repo/github.com/khusamov/solid-futurio-ru-game/node_modules/typescript-rtti/src/transformer/metadata-emitter.ts:38:23)
    at /Users/khusamov/Documents/repo/github.com/khusamov/solid-futurio-ru-game/node_modules/typescript-rtti/src/transformer/metadata-emitter.ts:87:41
    at MetadataEmitter.scope (/Users/khusamov/Documents/repo/github.com/khusamov/solid-futurio-ru-game/node_modules/typescript-rtti/src/transformer/metadata-emitter.ts:53:20)
    at MetadataEmitter.class (/Users/khusamov/Documents/repo/github.com/khusamov/solid-futurio-ru-game/node_modules/typescript-rtti/src/transformer/metadata-emitter.ts:86:21)
    at /Users/khusamov/Documents/repo/github.com/khusamov/solid-futurio-ru-game/node_modules/typescript-rtti/src/transformer/common/visitor-base.ts:62:44
    at visitNodes (/Users/khusamov/Documents/repo/github.com/khusamov/solid-futurio-ru-game/node_modules/typescript/lib/typescript.js:87753:48)
    at visitLexicalEnvironment (/Users/khusamov/Documents/repo/github.com/khusamov/solid-futurio-ru-game/node_modules/typescript/lib/typescript.js:87793:22)
```

```typescript
import {ICommand, IStoppable, IUniversalObject} from 'khusamov-base-types';
import IObjectWithStoppable from './IObjectWithStoppable';
import {resolve} from 'khusamov-inversion-of-control';
import {reflect} from 'typescript-rtti';

export default class StopCommand implements ICommand {
	private withStoppable: IObjectWithStoppable
	private stoppableCommand: ICommand & IStoppable | undefined

	get name(): string {
		return 'StopCommand: ' + (this.stoppableCommand?.name || `[not found '${this.stoppableCommandName}']`)
	}

	constructor(
		private stoppableCommandName: string,
		private targetObject: IUniversalObject
	) {
		this.withStoppable = resolve<IObjectWithStoppable>('Adapter', this.targetObject, reflect<IObjectWithStoppable>())
		this.stoppableCommand = this.withStoppable.stoppableCommandMap?.get(this.stoppableCommandName)
	}

	public execute(): void {
		this.stoppableCommand?.stop()
		this.withStoppable.stoppableCommandMap?.delete(this.stoppableCommandName)
	}
}
```

The error is caused by this line:

`this.withStoppable = resolve<IObjectWithStoppable>('Adapter', this.targetObject, reflect<IObjectWithStoppable>())`

Runtime error:

```
Uncaught ReferenceError: LΦ_0 is not defined
    at Object.LΦ (index.es.js:1641:36)
    at Object.a (index.es.js:1636:23)
    at new StopCommand (index.es.js:1661:136)
    at stopCommandResolver (index.es.js:1757:12)
    at IoC.resolve (index.es.js:2051:16)
    at Function.resolve (index.es.js:2041:42)
    at resolve (index.es.js:2269:16)
    at convertAgentMessage (index.es.js:1982:12)
    at AgentMessageInterpretCommand.execute (index.es.js:1998:34)
    at MacroCommand.execute (index.es.js:1372:21)
```

**Comment by rezonant** [comment]
Yep, this is the same as #54 which is fixed in dev but unreleased pending fixes needed to satisfy the corpus test suite

**Comment by rezonant** [comment]
Let's leave this open until the release, as the corpus failures were unanticipated and might take some time

**Comment by rezonant** [comment]
As a workaround, the issue is that call site reflection is used in the constructor. If you move the call site reflection somewhere else, it should solve the issue, but this may not work for your use case.

**Comment by rezonant** [comment]
This should be fixed in 0.6.0 which is now released.

**Comment by khusamov** [comment]
Thanks! The error is gone!
